### PR TITLE
CANS-1107 2.0 Print Functionality Implemented

### PR DESCRIPTION
**Pull request by zliuwd**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-1107

## Description

Rebuild a Print functionality that is matching 2.0 design
## Remarks
- New Print should not change the overall behavior of the current Print.

- All Print related components should come with the regular unit tests like other components in our app.

- A set of reusable components for print or a framework should be created in this story.

- In-line styling and CSS string are allowed using in this story. (In the plan, we use inline styling mostly for the customized components of every sub-functionality, and the CSS string is mainly for preparing Print container, reusable components, and some dynamically changing styling )

- px and percentage ( unit of measurement) are allowed using in this story. (Because safari does not support SVG which using 'rem' very well)

- use SVG to resolve that color could not be printed out under IE11

- the "water marker" in the design will not be implemented by this story(confirmed with design team), because of the approaching still not clear and will do in next sprint. 

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

Screen Shot 2019-04-11 at 11 28 47 PM

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Print/printAssessment/PrintAssessmentHeader.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+          <div style={headerDescriptionBox}>Child/youth has caregiver?</div>
+          <div />
+        </div>
+        <div style={headerThreeColsBox}>
+          <div style={headerContentBox}>
+            <PrintAssessmentHeaderOptions isChecked={Boolean(this.props.canReleaseInfo)} />
+          </div>
+          <div style={headerDescriptionBox}>Authorization for release of information on file?</div>
+          <div style={headerSubDescriptionBox}>
+            {this.props.canReleaseInfo || 'Items 8, 48 and EC 41 (Substance Abuse Items) are redacted.'}
```
Comment (LeonidMarushevskyi) [inline app/javascript/Application/components/Print/printAssessment/PrintAssessmentHeader.js]: This label should use substanceUseItemsIds. See comment above.

File: app/javascript/Application/components/Print/printAssessment/PrintAssessment.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        ) : null}
+      </PrintLayout>
+    )
+  }
+}
+
+PrintAssessment.propTypes = {
+  assessment: PropTypes.object.isRequired,
+  i18n: PropTypes.object.isRequired,
+  substanceUseItemsIds: PropTypes.shape({
```
Comment (LeonidMarushevskyi) [inline app/javascript/Application/components/Print/printAssessment/PrintAssessment.js]: Seems like you lost functionality related to substance use label being dynamic based on assessment structure. In some point, we switched substance use item ids 7 and 8 that is why we need to use this property. I see it is not used now.

File: app/javascript/Application/components/Print/PrintAssessment.js [hunk 3]
```diff
@@ -1,246 +0,0 @@
...
-        </div>
-      </div>
-    )
-  }
-
-  renderConfidentialWarningAlert = () => {
-    return (
-      <div style={alertSignBox}>
-        <strong>
-          By selecting NO, Items {this.props.substanceUseItemsIds.aboveSix[0]}, 48, and EC 41 (Substance Use Disorder
```
Comment (LeonidMarushevskyi) [inline app/javascript/Application/components/Print/PrintAssessment.js]: This is how it was before regarding 7/8 substance use item swap.

File: app/javascript/Application/components/Print/printAssessment/PrintAssessmentHeader.js (hunk 1 above)
Comment (zliuwd) [reply app/javascript/Application/components/Print/printAssessment/PrintAssessmentHeader.js]: done

**Pull request merged**